### DISCUSSION TITLE
Is Custom Authentication still possible after 3.0.0 Versions ?

### DISCUSSION THREAD

**Discussion by namsancow** [iTwin/itwinjs-core#3510] [Q&A] [answered]
Hi,

I just read the Q&A #2539 as followings
"How to replace Bentley authentication with custom authentication in iTwin?"

In the meantime, to use iTwin, I registered in iTwin Platform and then used iTwinId, iModelId, and authClient issued from this Platform.

However, if you look at Q&A #2539, the answer is that Custom Authentication is also possible. 
According to Q&A #2539, It is understood that the iTwin Platform registration process can be completely "Customized".

Is Custom Authentication possible in iTwin version 3.0?
Maybe it was possible in version 1.0 ~ 2.0, but isn't it possible after version 3.0?

if Custom Authetication is possible even in iTwin version 3.0, I wonder if it is possible iTwinId, iModelId, and authClient can all be issued by customization.

**Comment by calebmshafer** [comment, marked as answer]
Hi @namsancow, yes it is still possible with iTwin.js 3.x. It's been made a bit easier in some cases as well with the updates made during the release. All auth is handled using the [AuthorizationClient](https://www.itwinjs.org/reference/core-common/authorization/authorizationclient/) interface which has a minimal API contract.

The best example of how to do this is going to be [this repository](https://github.com/calebmshafer/itwin-viewer-proxy/) if you're building a web app.

**Reply by namsancow** [reply]
Many thanks for kind answer. 
iTwin's open ecosystem policy is truly the best !!!
